### PR TITLE
Timeouts, sockets and tests

### PR DESCRIPTION
**Pull request by mmalecki**
TODO: make those use @soldair's collector mock

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -40,14 +41,21 @@ function httpCheck(module, url, cb) {
 
 var checks = {
   tcp: function (url, cb) {
-    net.connect({
+    var timeout = setTimeout(function () {
+      socket.destroy()
+      cb(null, false, true)
+    }, DEFAULT_TIMEOUT);
```
Comment (chrisdickinson) [inline index.js:47]: Any reason not to use `socket.setTimeout(DEFAULT_TIMEOUT, function () {})` here?

(Ah, `clearTimeout`, probably!)

File: test/net-timeout-test.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      'tcp://127.0.0.1:' + TARGET_PORT
+    ],
+    interval: 100
+  })
+
+  collector.once('data',function(){
+    stopProducer()
+    collector.finished(function(err,metrics){
+
+      console.log(metrics)
```
Comment (chrisdickinson) [inline test/net-timeout-test.js]: might nix the `console.log` here.

**Comment by chrisdickinson** [comment]
Other than linting concerns which esformatter should be able to pick up, this LGTM! Great work!

**Comment by soldair** [comment]
thanks! yeah I need to push up the esformatter repo for numbat

**Pull request merged**